addons/account/models/account_bank_statement.py

```diff
@@ -17,6 +17,7 @@ def default_get(self, fields_list):
         if self._context.get('st_line_id'):
             st_line = self.env['account.bank.statement.line'].browse(self._context['st_line_id'])
             defaults['balance_start'] = st_line.running_balance - (st_line.amount if st_line.state == 'posted' else 0)
+            defaults['balance_end_real'] = st_line.running_balance
```

Comment:
for performances, we need to compute balance_start / balance_end_real only if those fields are in `fields_list`